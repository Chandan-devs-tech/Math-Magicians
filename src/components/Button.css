.btn {
  width: 80px;
  height: 80px;
  flex: 1;
  font-size: 25px;
  border: 1px solid lightgrey;
  cursor: pointer;
}

.zeroBtn {
  flex: 2;
}

.orangeColor {
  background: #ffa500;
}
