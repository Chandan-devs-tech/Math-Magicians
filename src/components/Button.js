import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Button.css';

function Button({ text, handleClick }) {
  const btnClass = text === '0' ? 'zeroBtn btn' : 'btn';
  const orangeColor = ['รท', '+', '=', '-', 'x'].includes(text) ? 'orangeColor' : '';
  return (
    <button type="button" className={`${btnClass} ${orangeColor}`} onClick={handleClick}>{text}</button>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
