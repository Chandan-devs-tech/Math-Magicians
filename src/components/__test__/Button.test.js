import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  test('should render the Button component', () => {
    const handleClick = jest.fn();
    const component = renderer.create(<Button text="5" handleClick={handleClick} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree = component.root;
    expect(tree.findByType(Button).props.text).toBe('5');
  });
});
