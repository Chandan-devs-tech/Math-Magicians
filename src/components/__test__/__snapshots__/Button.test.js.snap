// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button component 1`] = `
<button
  className="btn "
  onClick={[MockFunction]}
  type="button"
>
  5
</button>
`;
